runelite-client/src/main/java/net/runelite/client/plugins/thieving/ThievingPlugin.java

```diff
@@ -73,7 +86,11 @@
 	@Getter(AccessLevel.PACKAGE)
 	private ThievingSession session;
 
+	@Getter(AccessLevel.PACKAGE)
+	private final List<ChestRespawn> respawns = new ArrayList<>();
+
 	private int statTimeout;
+	private boolean recentlyLoggedIn = false;
```

Comment:
I feel like this entire logging in/out thing is more hassle than it's worth. The longest respawn takes less than 10 minutes, and I don't think there's going to be more than 2 people total doing this, at any given time. Clearing on log out should be enough

Reply:
I haven't test with thieving chest, but in mining plugin they wrote: 
https://github.com/open-osrs/runelite/blob/8b7eaf23f1b1b0e2b044110eeb209c7f6665f468/runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java#L171-L173

So I added the check to be safe.

Reply:
Oh, this is because of spawn events when regions load, I thought it had to do with keeping track of worlds and such, mb